### PR TITLE
Add single level of categorization #126089

### PR DESCRIPTION
**Pull request by rzhao271**
This PR affects #126089

It adds grouping to the settings editor extensions menu so that extension authors can now group their settings into multiple categories. 

The way the grouping works on the schema side is as follows:
- The contributes configuration object takes in an array of objects, rather than a single object
- Each array element can specify an `order` parameter so that the categories appear in a certain order
- This PR does **not** add support for sorting extension settings in a specific order
- See https://github.com/microsoft/vscode/blob/main/extensions/css-language-features/package.json for an example

The way the grouping works when displaying the settings is as follows:
- If there is only one category, the extension configuration title is used, rather than the extension name. For example, for https://github.com/amirmarmul/laravel-blade-vscode/blob/master/package.json, in the settings, we display "Blade Language" instead of "Laravel Blade"
- If there is more than one category, for each category, we display the category title, and for the group containing the categories, we use the extension's display name.

Here's a new picture showing how the titles look:

Picture showing changes in this PR

### CODE REVIEW

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts
Comment (roblourens) [inline src/vs/workbench/contrib/preferences/browser/settingsTree.ts:384]: This is interesting, on one hand we may not need a bunch of single-child nodes, but on the other, now some of the nodes under "Extensions" are the extension name, but some others are the section title, which feels inconsistent. I still won't be able to tell which extension some config sections come from

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts
Comment (rzhao271) [inline src/vs/workbench/contrib/preferences/browser/settingsTree.ts:366]: The `id` and `order` fields are not stored in `IConfiguration`, so this code retrieves them indirectly.
It probably wouldn't hurt to add them as optional fields into the interface, assuming this use case is fine.

**Comment by rzhao271** [comment]
Here's how the toc currently looks

Screenshot

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts
Comment (roblourens) [reply src/vs/workbench/contrib/preferences/browser/settingsTree.ts:381]: Yeah,  add it to whatever interface it needs to be on instead of looking it up this way

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts
Comment (roblourens) [inline src/vs/workbench/contrib/preferences/browser/settingsTree.ts:404]: This is fine but gitlens will lose the description that it puts in the section title. We can ask Eric how annoying that is, but if he adopts the section view by splitting his settings into sections, then he can keep that description on the top section.

**Review by roblourens — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/preferences/browser/media/settingsEditor2.css
Comment (rzhao271) [reply src/vs/workbench/contrib/preferences/browser/media/settingsEditor2.css:285]: It reduces the width of groups that use very long titles.
I could add that rule under a new selector such as `.monaco-tl-contents.group-title`, though.
Example showing long group title being cut off

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts
Comment (roblourens) [inline src/vs/workbench/contrib/preferences/browser/settingsTree.ts:382]: I guess I was thinking that we shouldn't be looking up the configuration off the extension contribution, since aren't we sort of re-looking up the configuration that we already have on ISettingsGroup? Should that have the order?

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts
Comment (roblourens) [reply src/vs/workbench/contrib/preferences/browser/settingsTree.ts:407]: So basically if you haven't already, talk to him to see whether he would want to adopt it right away, or if he's ok with losing the extra description until then, because we shouldn't merge this and surprise him by losing that message

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts
Comment (rzhao271) [reply src/vs/workbench/contrib/preferences/browser/settingsTree.ts:382]: I took a look and noticed `configurationRegistry.ts`. It seems `IConfigurationNode` has all the fields I need, but I'll have to see how that interface is being translated to `ISettingsGroup`, and how it's being populated in the first place. I set up a breakpoint and I think `order` wasn't preserved in the `IConfigurationNode` objects.

File: src/vs/workbench/contrib/preferences/browser/media/settingsEditor2.css
Comment (rzhao271) [inline src/vs/workbench/contrib/preferences/browser/media/settingsEditor2.css:292]: Testing GHPRI extension

**Comment by rzhao271** [comment]
Testing GHPRI

**Comment by rzhao271** [comment]
Testing extension

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts
Comment (rzhao271) [reply src/vs/workbench/contrib/preferences/browser/settingsTree.ts:400]: I changed the flattening case to use `value.children![0].label` now instead, which uses the same labels as in Insiders.

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts
Comment (rzhao271) [reply src/vs/workbench/contrib/preferences/browser/settingsTree.ts]: Got this fixed.

**Review by roblourens — APPROVED** [review]

**Pull request merged**
